frontend/pages/dataset/_id/annotation-mode/index.vue

```diff
@@ -56,7 +56,7 @@ export default {
       areResponsesUntouched: true, // NOTE - this flag is used to show or to not show a toast when questionnaire is touched (to prevent loosing current modification)
     };
   },
-  beforeRouteLeave(to, from, next) {
+  async beforeRouteLeave(to, from, next) {
     if (!this.areResponsesUntouched) {
```

Comment:
I think this could have been rewritten in a simpler way.
```javascript
if (!this.areResponsesUntouched && ["datasets", "dataseat-id-settings", "user-settings"].includes(to.name)) {
  this.showNotification({
    eventToFireOnClick: next,
    message: "Your changes will be lost if you leave the current page",
    buttonMessage: this.buttonMessage,
  });
} else {
   await next();
}